src/core.jl

```diff
@@ -75,6 +75,23 @@ struct Query <: AbstractQuery
     Query(str::AbstractString) = new(process_punct(str))
 end
 
+
+"""
+    Base.match(query::AbstractQuery, text::Union{Document,Corpus})
+
+Looks for a match for `query` in the text. Returns either `nothing`
+if no match is found, or a [`QueryMatch`](@ref) object.
+"""
+Base.match(query::AbstractQuery, text::Union{Document,Corpus})
```

Comment:
Maybe I'm missing something obvious, but where's the function definition?

Reply:
Yeah, so in all these I'm on-purpose not attaching the docstring to the specific methods, because the existing methods are more narrow than where I want the docstring to apply. E.g. here I want the docstring to apply to any method of `Base.match` that fits that signature (instead of sticking it on just one of the `match` methods, or copying it by hand to all of them). For the explain one, I want it to apply to any method of `explain`. 

Why does it matter? if for example you do `? +` then you get all the docstrings for any method of `+`. But if you do `? +(1,1)`, you get just 1 docstring, the most specific one that applies to those arguments.

Reply:
😮  TIL!